vmms/dockerSSH.py

```diff
@@ -0,0 +1,221 @@
+import config
+from tangoObjects import TangoMachine
+
+def timeout(command, time_out=1):
+    """ timeout - Run a unix command with a timeout. Return -1 on
+    timeout, otherwise return the return value from the command, which
+    is typically 0 for success, 1-255 for failure.
+    """ 
+
+    # Launch the command
+    p = subprocess.Popen(command,
+                        stdout=open("/dev/null", 'w'),
+                        stderr=subprocess.STDOUT)
+
+    # Wait for the command to complete
+    t = 0.0
+    while t < time_out and p.poll() is None:
+        time.sleep(config.Config.TIMER_POLL_INTERVAL)
+        t += config.Config.TIMER_POLL_INTERVAL
+
+    # Determine why the while loop terminated
+    if p.poll() is None:
+        subprocess.call(["/bin/kill", "-9", str(p.pid)])
+        returncode = -1
+    else:
+        returncode = p.poll()
+    return returncode
+
+def timeoutWithReturnStatus(command, time_out, returnValue = 0):
+    """ timeoutWithReturnStatus - Run a Unix command with a timeout,
+    until the expected value is returned by the command; On timeout,
+    return last error code obtained from the command.
+    """
+    p = subprocess.Popen(command, 
+                        stdout=open("/dev/null", 'w'), 
+                        stderr=subprocess.STDOUT)
+    t = 0.0
+    while (t < time_out):
+        ret = p.poll()
+        if ret is None:
+            time.sleep(config.Config.TIMER_POLL_INTERVAL)
+            t += config.Config.TIMER_POLL_INTERVAL
+        elif ret == returnValue:
+            return ret
+        else:
+            p = subprocess.Popen(command,
+                            stdout=open("/dev/null", 'w'),
+                            stderr=subprocess.STDOUT)
+    return ret
+
+#
+# User defined exceptions
+#
+
+class DockerSSH:
+    _OS_X = 'darwin'
+    LOCALHOST = '127.0.0.1'
+
+    def __init__(self):
+        """ Checks if the machine is ready to run docker containers.
+        Initialize boot2docker if running on OS X.
+        """
+        try:
+            self.log = logging.getLogger("DockerSSH")
+            # If running on OS X, create a boot2docker VM
+            if sys.platform == self._OS_X:
+                # boot2docker initialization will be part of initial
+                # set up with Tango.
+                self.docker_host_ip = subprocess.check_output(['boot2docker', 'ip']).strip('\n')
+            else:
+                self.docker_host_ip = self.LOCALHOST
+
+            # Check import docker constants are defined in config
+            if len(config.Config.DOCKER_VOLUME_PATH) == 0:
+                raise Exception('DOCKER_VOLUME_PATH not defined in config.')
+
+            if len(config.Config.DOCKER_IMAGE) == 0:
+                raise Exception('DOCKER_IMAGE not defined in config.')
+
+            self.log.info("Docker host IP is %s" % self.docker_host_ip)
+
+        except Exception as e:
+            self.log.error(str(e))
+            exit(1)
```

Comment:
Well, it logs the error. We could also print the error to stdout. But it should die.